Bottom: fd757ff2ef7cf8485c3e558d91dc74afaaec9617
Top:    f3fee5b83c0303a36e67ed9e9d201c1103578e9e
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 18:26:13 +0000

Fix board outline generation to work when we find nothing

This was hit due to another bug, but since NULL is a valid polygon, we
should handle this case!


---

diff --git a/src/polygon.c b/src/polygon.c
index be4fcf2..7be99db 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -2462,8 +2462,8 @@ POLYAREA *board_outline_poly (bool include_holes)
   /* Now we just need to work out which pieces of polygon are inside
      and outside the board! */
 
-  /* If there is only one piece, return that */
-  if (clipped->f == clipped)
+  /* If there is no result, or only one piece, return that */
+  if (clipped == NULL || clipped->f == clipped)
     return clipped;
 
   /* WARNING: This next check is O(n^2), where n is the number of clipped
